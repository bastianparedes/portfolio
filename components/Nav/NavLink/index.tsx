import React, { MouseEventHandler } from 'react'
import styles from './styles.module.css'

interface NavLinkProps {
  link: string
  title: string
  onClick: MouseEventHandler<HTMLAnchorElement>
}

const NavLink = (props: NavLinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a
        href={'#' + props.link}
        className={styles.link__a}
        onClick={props.onClick}
      >
        {props.title}
      </a>
    </li>
  )
}

export default NavLink
