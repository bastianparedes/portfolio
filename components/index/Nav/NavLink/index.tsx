import React from 'react';

import styles from './styles.module.scss';

interface NavLinkProps {
  link: string;
  title: string;
  closeNav: () => void;
}

const NavLink = ({ link, title, closeNav }: NavLinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a href={'#' + link} className={styles.link__a} onClick={closeNav}>
        {title}
      </a>
    </li>
  );
};

export default NavLink;
