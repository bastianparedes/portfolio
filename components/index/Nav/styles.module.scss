@import 'bastianparedes/styles/_breakpoints';

.nav {
  height: fit-content;
  background-color: rgb(0, 16, 85);
  width: 100%;
  position: sticky;
  top: 0;
  display: grid;
  grid-template-columns: 1fr 1fr;
  z-index: 3;
}

.navLeft {
  grid-column: 1;
  grid-row: 1;
  height: 4rem;
  position: relative;
  object-fit: cover;
  display: flex;
  align-items: center;
}

.logoContainer {
  width: 116.63px;
  height: 51.2px;
  position: relative;
  margin-left: 1rem;
}

.buttonContainer {
  grid-column: 2;
  grid-row: 1;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.button {
  width: 3.5rem;
  height: 3.5rem;
  font-size: 2.7rem;
  display: none;
  align-items: center;
  justify-content: center;
  border: none;
  margin-right: 1rem;
  cursor: pointer;
  transition: all ease 1s;
  z-index: 1;
}

.buttonAnimation {
  transform: rotate(180deg);
}

.linksContainer {
  grid-column: 2;
  grid-row: 1;
  width: fit-content;
  list-style: none;
  margin: 0 3rem 0 auto;
  display: flex;
  align-items: center;
  gap: 3rem;
  font-size: 2rem;
  background-color: inherit;
}

@include phone-and-tablet {
  .button {
    display: flex;
  }

  .linksContainer {
    width: 100%;
    grid-column: 1 / 3;
    grid-row: 2;
    position: absolute;
    display: grid;
    justify-content: center;
    align-items: center;
    grid-template-columns: 1fr;
    grid-auto-rows: 1fr;
    gap: 0;
    margin: 0rem 0;
    padding: 0;
    border-color: rgb(133, 133, 133);
    border-width: 3px;
    border-style: solid none;
    transition: all linear 0.5s;
    transform: scaleY(0);
    transform-origin: top;
  }

  .linksContainerOpened {
    transform: scaleY(1);
  }
}
