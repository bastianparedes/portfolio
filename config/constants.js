const constants = {
  ABOUT: {
    CV: {
      LINK: '/files/cv.pdf',
      MESSAGE: 'Ver currículum'
    },
    DESCRIPTION_PART_1: 'Soy Bastián Paredes,',
    DESCRIPTION_PART_2:
      'Un Front End Developer con experiencia construyendo sitios web con responsive design que conecten front-end y back-end. Me especializo en el uso de JavaScript, React.js y Next.js.',
    DESCRIPTION_PART_3:
      'Siempre busco mejorar mis habilidades desarrollando proyectos propios a la par de mi trabajo. Estoy constantemente estudiando acerca de las nuevas tecnologías para actualizar mi repertorio de herramientas para que los códigos de mis creaciones sean fáciles de leer, seguros y eficientes.',
    SECTION: {
      TITLE: 'About me'
    },
    SKILLS: {
      LIST: [
        {
          NAME: 'HTML 5',
          SRC: 'html5.png'
        },
        {
          NAME: 'CSS 3',
          SRC: 'css3.png'
        },
        {
          NAME: 'JavaScript',
          SRC: 'javascript.png'
        },
        {
          NAME: 'TypeScript',
          SRC: 'typescript.png'
        },
        {
          NAME: 'React.js',
          SRC: 'react.js.png'
        },
        {
          NAME: 'Next.js',
          SRC: 'next.js.png'
        },
        {
          NAME: 'Python 3',
          SRC: 'python3.png'
        },
        {
          NAME: 'MySql',
          SRC: 'mysql.png'
        },
        {
          NAME: 'MongoDB',
          SRC: 'mongodb.svg'
        }
      ],
      TITLE: 'Skills'
    }
  },
  CONTACT: {
    LINKS: {
      EMAIL: '/contact',
      GITHUB: 'https://github.com/BastianParedes',
      INSTAGRAM: 'https://www.instagram.com/bastian.p.95/',
      LINKEDIN: 'https://www.linkedin.com/in/bastianparedes/'
    }
  },
  HOME: {
    JOB: 'Front End Developer',
    NAME: 'Bastián Paredes'
  },
  NAV: {
    BUTTONS: {
      ABOUT: {
        ID: 'about',
        TITLE: 'About'
      },
      HOME: {
        ID: 'home',
        TITLE: 'Home'
      },
      PORTFOLIO: {
        ID: 'portfolio',
        TITLE: 'Porftolio'
      }
    }
  },
  PORTFOLIO: {
    PROJECTS: [
      {
        DESCRIPTION:
          'A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.',
        IMAGE: 'falabella.png',
        LINK: 'https://www.falabella.com/falabella-cl/',
        NAME: 'Falabella',
        TECHNOLOGIES: [
          'HTML',
          'CSS',
          'JavaScript',
          'React.js',
          'Next.js',
          'TypeScript',
          'Dinamic Yield',
          'Adobe Experience Data Collection'
        ]
      },
      {
        DESCRIPTION:
          'Página web de venta de bowls de ensalada que incluye un carrito de compras y una pasarela de pago construida mediante el servicio Webpay plus de Transbank. Las compras quedan registradas en una base de datos para consultar su estado.',
        GITHUB: 'https://github.com/BastianParedes/tusalad/',
        IMAGE: 'tu-salad.png',
        LINK: '/tusalad',
        NAME: 'Tu Salad',
        TECHNOLOGIES: ['React.js', 'Next.js', 'TypeScript', 'WebPay']
      },
      {
        DESCRIPTION:
          'Es una página web que recibe las imágenes cargadas por el usuario, las configura a gusto y finalmente genera un archivo PDF que se guarda en el computador del mismo. Todo el proceso ocurre del lado del Front-end.',
        GITHUB: 'https://github.com/BastianParedes/pdf-constructor/',
        IMAGE: 'pdf-constructor.png',
        LINK: '/pdf-constructor',
        NAME: 'PDF Constructor',
        TECHNOLOGIES: ['React.js', 'SortableJS', 'jsPDF']
      },
      {
        DESCRIPTION:
          'Aplicación web que pide al usuario un nombre y lo lleva a una sala donde todos los conectados a la página en ese momento pueden conversar al mismo tiempo.',
        GITHUB: 'https://github.com/BastianParedes/my-global-chat/',
        IMAGE: 'my-global-chat.png',
        LINK: '/my-global-chat',
        NAME: 'My Global Chat',
        TECHNOLOGIES: ['React.js', 'Socket.IO']
      },
      {
        DESCRIPTION:
          'CRUD que administra las contraseñas ingresadas por el ususario, las encripta y la muestra en el Front-End.',
        GITHUB: 'https://github.com/BastianParedes/passwords/',
        IMAGE: 'passwords.png',
        LINK: '/passwords',
        NAME: 'Passwords',
        TECHNOLOGIES: ['Next.js', 'MySQL', 'Prisma', 'Bcrypt']
      },
      {
        DESCRIPTION:
          'Aplicación que genera el código QR correspondiente a la URL ingresada.',
        GITHUB: 'https://github.com/BastianParedes/qr-generator/',
        IMAGE: 'qr-generator.png',
        LINK: '/qr-generator',
        NAME: 'QR generator',
        TECHNOLOGIES: ['QRcode', 'Next.js']
      },
      {
        DESCRIPTION:
          'Aplicación que muestra el clima actual en la ubicación ingresada.',
        GITHUB: 'https://github.com/BastianParedes/weather/',
        LINK: '/weather',
        NAME: 'Weather',
        TECHNOLOGIES: ['Google Maps API', 'openWeather API', 'Next.js']
      },
      {
        DESCRIPTION:
          'Aplicación de escritorio que construye evaluaciones de seleción única de matemática para el nivel secundario de las escuelas chilenas.',
        GITHUB: 'https://github.com/BastianParedes/tstmaker/',
        IMAGE: 'tstmaker.png',
        NAME: 'TstMaker',
        TECHNOLOGIES: ['Python3', 'Tkinter', 'MiKTeX']
      }
    ]
  }
};

export default constants;
