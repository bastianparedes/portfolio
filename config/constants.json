{
  "HOME": {
    "NAME": "Bastián Paredes",
    "JOB": "Front End Developer"
  },
  "NAV": {
    "BUTTONS": {
      "HOME": {
        "ID": "home",
        "TITLE": "Home"
      },
      "ABOUT": {
        "ID": "about",
        "TITLE": "About"
      },
      "PORTFOLIO": {
        "ID": "portfolio",
        "TITLE": "Porftolio"
      }
    }
  },
  "ABOUT": {
    "SECTION": {
      "TITLE": "About me"
    },
    "DESCRIPTION_PART_1": "Soy Bastián Paredes,",
    "DESCRIPTION_PART_2": "Un Front End Developer con experiencia construyendo sitios web con responsive design que conecten front-end y back-end. Me especializo en el uso de JavaScript, React.js y Next.js.",
    "DESCRIPTION_PART_3": "Siempre busco mejorar mis habilidades desarrollando proyectos propios a la par de mi trabajo. Estoy constantemente estudiando acerca de las nuevas tecnologías para actualizar mi repertorio de herramientas para que los códigos de mis creaciones sean fáciles de leer, seguros y eficientes.",
    "CV": {
      "MESSAGE": "Ver currículum",
      "LINK": "/api/cv"
    },
    "SKILLS": {
      "TITLE": "Skills",
      "LIST": [
        {
          "SRC": "html5.png",
          "NAME": "HTML 5"
        },
        {
          "SRC": "css3.png",
          "NAME": "CSS 3"
        },
        {
          "SRC": "javascript.png",
          "NAME": "JavaScript"
        },
        {
          "SRC": "typescript.png",
          "NAME": "TypeScript"
        },
        {
          "SRC": "react.js.png",
          "NAME": "React.js"
        },
        {
          "SRC": "next.js.png",
          "NAME": "Next.js"
        },
        {
          "SRC": "python3.png",
          "NAME": "Python 3"
        },
        {
          "SRC": "mysql.png",
          "NAME": "MySql"
        },
        {
          "SRC": "mongodb.svg",
          "NAME": "MongoDB"
        }
      ]
    }
  },
  "PORTFOLIO": {
    "PROJECTS": [
      {
        "NAME": "Falabella",
        "LINK": "https://www.falabella.com/falabella-cl/",
        "IMAGE": "falabella.png",
        "GITHUB": "",
        "DESCRIPTION": "A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.",
        "TECHNOLOGIES": [
          "HTML",
          "CSS",
          "JavaScript",
          "React.js",
          "Next.js",
          "TypeScript",
          "Dinamic Yield",
          "Adobe Experience Data Collection"
        ]
      },
      {
        "NAME": "Tu Salad",
        "LINK": "https://tusalad.vercel.app/",
        "IMAGE": "tu-salad.png",
        "GITHUB": "https://github.com/BastianParedes/tusalad/",
        "DESCRIPTION": "Página web de venta de bowls de ensalada que incluye un carrito de compras y una pasarela de pago construida mediante el servicio Webpay plus de Transbank. Las compras quedan registradas en una base de datos para consultar su estado.",
        "TECHNOLOGIES": ["React.js", "Next.js", "TypeScript", "WebPay"]
      },
      {
        "NAME": "PDF Constructor",
        "LINK": "https://pdf-constructor.vercel.app/",
        "IMAGE": "pdf-constructor.png",
        "GITHUB": "https://github.com/BastianParedes/pdf-constructor/",
        "DESCRIPTION": "Es una página web que recibe las imágenes cargadas por el usuario, las configura a gusto y finalmente genera un archivo PDF que se guarda en el computador del mismo. Todo el proceso ocurre del lado del Front-end.",
        "TECHNOLOGIES": ["React.js", "SortableJS", "jsPDF"]
      },
      {
        "NAME": "My Global Chat",
        "LINK": "https://my-global-chat.vercel.app/",
        "IMAGE": "my-global-chat.png",
        "GITHUB": "https://github.com/BastianParedes/my-global-chat/",
        "DESCRIPTION": "Aplicación web que pide al usuario un nombre y lo lleva a una sala donde todos los conectados a la página en ese momento pueden conversar al mismo tiempo.",
        "TECHNOLOGIES": ["React.js", "Socket.IO"]
      },
      {
        "NAME": "TstMaker",
        "LINK": "",
        "IMAGE": "tstmaker.png",
        "GITHUB": "https://github.com/BastianParedes/tstmaker/",
        "DESCRIPTION": "Aplicación de escritorio que construye evaluaciones de seleción única de matemática para el nivel secundario de las escuelas chilenas.",
        "TECHNOLOGIES": ["Python3", "Tkinter", "MiKTeX"]
      }
    ]
  },
  "CONTACT": {
    "LINKS": {
      "GITHUB": "https://github.com/BastianParedes",
      "EMAIL": "/contact",
      "LINKEDIN": "https://www.linkedin.com/in/bastianparedes/",
      "INSTAGRAM": "https://www.instagram.com/bastian.p.95/"
    }
  },
  "FOOTER": {
    "MESSAGE": "Portfolio designed by @Bastián Paredes, © 2022"
  }
}
