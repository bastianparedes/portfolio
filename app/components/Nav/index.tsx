'use client';

import React from 'react';

import { classNames } from 'bastianparedes/utils';
import Image from 'next/image';
import { AiOutlineMenu } from 'react-icons/ai';

import NavLink from './NavLink';
import styles from './styles.module.scss';
import constants from '../../../config/constants';
import logo from '../../../public/images/logo.png';

const Nav = (): JSX.Element => {
  const [opened, setOpened] = React.useState(false);

  const buttonClickHandler = (): void => {
    setOpened((value) => !value);
  };

  const closeNav = (): void => {
    setOpened(false);
  };

  return (
    <nav className={styles.nav}>
      <div className={styles.navLeft}>
        <div className={styles.logoContainer}>
          <Image
            alt={'BastiÃ¡n Paredes Logo'}
            placeholder="blur"
            src={logo}
            style={{
              height: '100%',
              objectFit: 'contain',
              width: '100%'
            }}
          />
        </div>
      </div>
      <div className={styles.buttonContainer}>
        <button
          className={classNames(
            styles.button,
            opened && styles.buttonAnimation
          )}
          data-testid="data-testid-nav__button"
          onClick={buttonClickHandler}
        >
          <AiOutlineMenu />
        </button>
      </div>
      <ul
        className={classNames(
          styles.linksContainer,
          opened && styles.linksContainerOpened
        )}
        data-testid="data-testid-nav__ul"
      >
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.HOME.ID}
          title={constants.NAV.BUTTONS.HOME.TITLE}
        />
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.ABOUT.ID}
          title={constants.NAV.BUTTONS.ABOUT.TITLE}
        />
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.PORTFOLIO.ID}
          title={constants.NAV.BUTTONS.PORTFOLIO.TITLE}
        />
      </ul>
    </nav>
  );
};

export default Nav;
