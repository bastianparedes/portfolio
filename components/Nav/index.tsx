import React from 'react'
import { AiOutlineMenu } from 'react-icons/ai'
import styles from './styles.module.css'
import constants from '../../config/constants.json'
import Image from 'next/image'
import NavLink from './NavLink'
import classNames from '../../utils/classNames'

const Nav = (): JSX.Element => {
  const [opened, setOpened] = React.useState(false)

  const buttonClickHandler = (): void => {
    setOpened((value) => !value)
  }

  const closeNav = (): void => {
    setOpened(false)
  }

  return (
    <nav className={styles.nav}>
      <div className={styles.navLeft}>
        <div className={styles.logoContainer}>
          <Image
            src={'/images/logo.png'}
            alt="BastiÃ¡n Paredes Logo"
            layout="fill"
            objectFit="contain"
          />
        </div>
      </div>
      <div className={styles.buttonContainer}>
        <button
          className={classNames(
            styles.button,
            opened && styles.buttonAnimation
          )}
          data-testid="data-testid-nav__button"
          onClick={buttonClickHandler}
        >
          <AiOutlineMenu />
        </button>
      </div>
      <ul
        className={classNames(
          styles.linksContainer,
          opened && styles.linksContainerOpened
        )}
      >
        <NavLink
          link={constants.NAV.BUTTONS.HOME.ID}
          title={constants.NAV.BUTTONS.HOME.TITLE}
          onClick={closeNav}
        />
        <NavLink
          link={constants.NAV.BUTTONS.ABOUT.ID}
          title={constants.NAV.BUTTONS.ABOUT.TITLE}
          onClick={closeNav}
        />
        <NavLink
          link={constants.NAV.BUTTONS.PORTFOLIO.ID}
          title={constants.NAV.BUTTONS.PORTFOLIO.TITLE}
          onClick={closeNav}
        />
      </ul>
    </nav>
  )
}

export default Nav
