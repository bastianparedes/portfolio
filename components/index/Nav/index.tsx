import React from 'react';

import { classNames } from 'bastianparedes/utils';
import Image from 'next/legacy/image';
import { AiOutlineMenu } from 'react-icons/ai';

import NavLink from './NavLink';
import styles from './styles.module.scss';
import constants from '../../../config/constants';

const Nav = (): JSX.Element => {
  const [opened, setOpened] = React.useState(false);

  const buttonClickHandler = (): void => {
    setOpened((value) => !value);
  };

  const closeNav = (): void => {
    setOpened(false);
  };

  return (
    <nav className={styles.nav}>
      <div className={styles.navLeft}>
        <div className={styles.logoContainer}>
          <Image
            alt="BastiÃ¡n Paredes Logo"
            layout="fill"
            objectFit="contain"
            placeholder="blur"
            src="/images/logo.png"
          />
        </div>
      </div>
      <div className={styles.buttonContainer}>
        <button
          className={classNames(
            styles.button,
            opened && styles.buttonAnimation
          )}
          data-testid="data-testid-nav__button"
          onClick={buttonClickHandler}
        >
          <AiOutlineMenu />
        </button>
      </div>
      <ul
        className={classNames(
          styles.linksContainer,
          opened && styles.linksContainerOpened
        )}
        data-testid="data-testid-nav__ul"
      >
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.HOME.ID}
          title={constants.NAV.BUTTONS.HOME.TITLE}
        />
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.ABOUT.ID}
          title={constants.NAV.BUTTONS.ABOUT.TITLE}
        />
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.PORTFOLIO.ID}
          title={constants.NAV.BUTTONS.PORTFOLIO.TITLE}
        />
      </ul>
    </nav>
  );
};

export default Nav;
