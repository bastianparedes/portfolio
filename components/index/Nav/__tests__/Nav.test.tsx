import React from 'react'
import Nav from '..'
import { render, screen, fireEvent } from '@testing-library/react'
import constants from '../../../../config/constants.json'

describe('<Nav />', () => {
  it('should render', () => {
    const { container } = render(<Nav />)
    expect(container).toMatchSnapshot()
  })

  it('should open nav and closses when link clicked', () => {
    render(<Nav />)
    const button = screen.getByTestId('data-testid-nav__button')
    const link = screen.getByText(constants.NAV.BUTTONS.HOME.TITLE)
    fireEvent.click(link)
    fireEvent.click(button)
    expect(screen).toMatchSnapshot()
  })
})
