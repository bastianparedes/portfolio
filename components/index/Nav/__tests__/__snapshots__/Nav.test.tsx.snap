// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should open nav and closses when link clicked 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`<Nav /> should render 1`] = `
<div>
  <nav
    class="nav"
  >
    <div
      class="navLeft"
    >
      <div
        class="logoContainer"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="BastiÃ¡n Paredes Logo"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
          />
          <noscript />
        </span>
      </div>
    </div>
    <div
      class="buttonContainer"
    >
      <button
        class="button"
        data-testid="data-testid-nav__button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </button>
    </div>
    <ul
      class="linksContainer"
    >
      <li
        class="link"
      >
        <a
          class="link__a"
          href="#home"
        >
          Home
        </a>
      </li>
      <li
        class="link"
      >
        <a
          class="link__a"
          href="#about"
        >
          About
        </a>
      </li>
      <li
        class="link"
      >
        <a
          class="link__a"
          href="#portfolio"
        >
          Porftolio
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
